tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1554,6 +1591,62 @@ private[optimizer] abstract class OptimizerCore(
       case LongToOctalStr =>
         contTree(Apply(firstArgAsRTLong, LongImpl.toOctalString, Nil)(StringClassType))
 
+      // scala.collection.mutable.ArrayBuilder
+
+      case GenericArrayBuilderResult =>
+        val List(runtimeClass, array) = newArgs
+        val (resultType, isExact) = runtimeClass match {
+          case ClassOf(elemType) => (ArrayType(elemType), true)
+          case _                 => (AnyType, false)
+        }
+        cont(PreTransTree(CallHelper("makeNativeArrayWrapper",
+            CallHelper("arrayDataOf",
+                CallHelper("classDataOf", runtimeClass)(AnyType))(AnyType),
```

Comment:
[Even the library knows that class data exist.](https://github.com/scala-js/scala-js/blob/master/javalanglib/src/main/scala/java/lang/Class.scala)